src/Instagram.php

```diff
@@ -525,16 +548,14 @@ public function pagination($obj, $limit = 0)
             }
 
             $function = str_replace(self::API_URL, '', $apiCall[0]);
+            $count = ($limit) ? $limit : count($obj->data);
 
-            $auth = (strpos($apiCall[1], 'access_token') !== false);
-
-            if (isset($obj->pagination->next_max_id)) {
-                return $this->_makeCall($function, $auth, array('max_id' => $obj->pagination->next_max_id, 'count' => $limit));
+            if (isset($obj->pagination->next_max_tag_id)) {
+                return $this->_makeCall($function, array('max_tag_id' => $obj->pagination->next_max_tag_id, 'count' => $count));
             }
 
-            return $this->_makeCall($function, $auth, array('cursor' => $obj->pagination->next_cursor, 'count' => $limit));
+            return $this->_makeCall($function, array('next_max_id' => $obj->pagination->next_max_id, 'count' => $count));
```

Comment:
I don't get any error from Instagram, but each pagination call returns the same batch of results. Changing "next_max_id" to "max_id" allowed me to get the next batch of items in the paginated list.